apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/api/CompilerParams.kt

```diff
@@ -193,4 +194,29 @@ interface CompilerParams {
    * the client was compiled against an older schema that doesn't have knowledge of the new enums.
    */
   val sealedClassesForEnumsMatching: ListProperty<String>
+
+  /**
+   * Whether or not to generate Apollo metadata. Apollo metadata is used for multi-module support. Set this to true if you want other
+   * modules to be able to re-use fragments and types from this module.
+   *
+   * This is currently experimental and this API might change in the future.
+   *
+   * Default value: false
+   */
+  @ApolloExperimental
+  val generateApolloMetadata: Property<Boolean>
```

Comment:
I expect multi-module to stay in experimental for some time so that we can tweak the APIs if needed.